lesson21/index.js

```diff
@@ -110,5 +112,70 @@ const createTd = (usersData, keys) => {
   return fragment;
 }
 
+const createSortBtn = () => {
+  const sortBtn = createElementWithClassName("button", "sort-btn");
+  sortBtn.dataset.sortStatus = "default";
+  sortBtn.id = "js-sort-btn";
+  return sortBtn;
+}
+
+const setClickInSortBtn = () => {
+  const defaultRows = [...document.querySelector("tbody").querySelectorAll("tr")];
+  const sortBtn = document.getElementById("js-sort-btn");
+
+  sortBtn.addEventListener("click", (e) => {
+    const nextStatus = switchSortStatus(e.target.dataset.sortStatus);
+    e.target.dataset.sortStatus = nextStatus;
+
+    const sortedRows = getSortedRows(nextStatus, defaultRows,e.target);
+
+    const tbody = document.querySelector("tbody");
+    sortedRows.forEach((row) => {
+      tbody.appendChild(row);
+    })
+  });
+}
+
+const switchSortStatus = (status) => {
+  switch (status) {
+    case "default":
+      return "asc";
+
+    case "desc":
+      return "default";
+
+    case "asc":
+      return "desc";
+
+    default:
+      return "default";
+  }
+}
+
+const findClickedCellIndex = (target) => {
+  const th = [...document.querySelector("thead").querySelectorAll("th")];
+  return th.indexOf(target.parentElement);
+}
+
+const getSortedRows = (status, defaultLows,target) => {
```

Comment:
```suggestion
const getSortedRows = (status, defaultRows,target) => {
```

Reply:
I'm sorry.
I let your precious time be used for my spelling mistakes. I'll be careful.

6c42d15